### PR TITLE
UI and Backend for private-events

### PR DESCRIPTION
**Pull request by Miguelus373**
## What this PR does:
- Adds all the MVC for this page
- Adds all the associations between the models
- Adds all the validations for the models
- Adds user interface to the page

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+  end
```
Comment (kobiyoyo) [inline app/controllers/events_controller.rb]: ##### EventsController
###### `index` action

* [x] lists past events ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  # GET /users
+  # GET /users.json
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  # GET /users/1.json
+  def show; end
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb:10]: ##### UsersController

###### `show` action

* [x] lists events created by user([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [x] Let's include a screenshot in the readme.MD file.
- [x] The Private Event title in the readme.MD file should start with a capital letter 
- [x] Let's remove all JSON formats.
- [x] View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
- [x] Let's use only Rspec and capybara for testing ,I can see a `test` DIR which is used when using Minitest, remove it .
* [x] authentication has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [x] event managment features has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
### [OPTIONAL]
### Routes
* [x] routes only for used actions, i.e. `resources :events, only: [:new, :create, :show, :index]`
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (happiguru) [reply app/controllers/users_controller.rb:10]: Resolved

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (happiguru) [reply app/controllers/events_controller.rb]: Resolved

**Review by christianotieno — APPROVED** [review]
### Project Approved 🍾 

Hello @Miguelus373 & @happiguru 👋 
Congratulations on the hard work with the changes. keep it up 👏 
All the required changes have been fulfilled in this project. I will go ahead and mark it as complete 👍 
I wish you all the best in the next tasks 🥇 

### Happy coding 💻 

#### Christian Otieno (TSE)

**Pull request merged**